src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -291,10 +291,7 @@ public static int GetGeneration(WeakReference wo)
             object? obj = GCHandle.InternalGet(wo.WeakHandle);
             KeepAlive(wo);
 
-            if (obj is null)
-            {
-                throw new ArgumentNullException(nameof(wo));
-            }
+            ArgumentNullException.ThrowIfNull(obj, nameof(wo));
```

Comment:
```suggestion
            ArgumentNullException.ThrowIfNull(obj);
```

Can be simplified further

Reply:
That's a behavioral change; it would change the name that's included in a resulting exception.